SDL/Segment.cu

```diff
@@ -918,27 +922,68 @@ __global__ void SDL::createSegmentsInGPUv2(struct SDL::modules& modulesInGPU, st
     int blockxSize = blockDim.x*gridDim.x;
     int blockySize = blockDim.y*gridDim.y;
     int blockzSize = blockDim.z*gridDim.z;
+
+    __shared__ short subdets[MAX_CONNECTED_MODULES], layers[MAX_CONNECTED_MODULES], sides[MAX_CONNECTED_MODULES], rods[MAX_CONNECTED_MODULES], rings[MAX_CONNECTED_MODULES], innerSubdet, innerLayer, innerSide, innerRod, innerRing;
```

Comment:
would performance get any better if full words are used here (32 bit, as `unsigned int`)?
My reading of https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#shared-memory-5-x suggests that the bank conflicts appear only for separate reads inside the bank.